com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/modelinference/InferredModelAssociator.java

```diff
@@ -56,15 +60,20 @@
    * An adapter that holds the mapping between source- and inferred-model elements.
    */
   public static class Adapter extends AdapterImpl {
-    private final ListMultimap<EObject, EObject> sourceToInferredModelMap = LinkedListMultimap.create();
+    private final Map<EObject, Set<EObject>> sourceToTargetMap = Maps2.newLinkedHashMapWithExpectedSize(40);
+    private final Map<EObject, Set<EObject>> targetToSourceMap = Maps2.newLinkedHashMapWithExpectedSize(40);
 
     @Override
     public boolean isAdapterForType(final Object type) {
       return Adapter.class == type;
     }
 
-    public ListMultimap<EObject, EObject> getSourceToInferredModelMap() {
-      return sourceToInferredModelMap;
+    public Map<EObject, Set<EObject>> getSourceToTargetMap() {

```

Comment:
i liked the previous name a bit more...

Reply:
I can change it back. I changed it so that the code would look more like the JvmModelAssociations class, in case we want to compare them again in the future.